src/test/java/ch/jalu/collectionbehavior/verification/MapMutabilityVerifier.java

```diff
@@ -0,0 +1,142 @@
+        assertThat(map, equalTo(Map.of("A", 1, "B", 4, "C", 9, "D", 4)));
+        map.replaceAll((k, v) -> v * 2);
+
+        // #remove
+        map.remove("C");
+
+        // #merge
+        map.merge("B", 1, Integer::sum);
+        assertThat(map, equalTo(Map.of("A", 2, "B", 9, "D", 8)));
+
+        // #compute, #computeIfPresent, #computeIfAbsent
+        map.compute("B", (k, v) -> null);
+        map.computeIfPresent("A", (k, v) -> v + 1);
+        map.computeIfPresent("Z", (k, v) -> v + 1); // no change
+        map.computeIfAbsent("D", k -> (int) k.charAt(0)); // no change
+        map.computeIfAbsent("E", k -> (int) k.charAt(0));
+        assertThat(map, equalTo(Map.of("A", 3, "D", 8, "E", 69)));
+
+        // #clear
+        map.clear();
+        assertThat(map, anEmptyMap());
+    }
+
+    public static void verifyMapKeySetIsMutable(Map<String, Integer> map) {
+        assertThat(map, anEmptyMap()); // Validate method contract
+        map.putAll(Map.of("a", 1, "b", 2, "c", 3, "d", 4));
+
+        Set<String> keySet = map.keySet();
+
+        keySet.remove("c");
+        keySet.removeIf(k -> k.equals("b"));
+        keySet.retainAll(Set.of("d", "z"));
+        assertThat(map, equalTo(Map.of("d", 4)));
+
+        keySet.clear();
+        assertThat(map, anEmptyMap());
+        assertThrows(UnsupportedOperationException.class, () -> keySet.add("f"));
+    }
+
+    public static void verifyMapValuesIsMutable(Map<String, Integer> map) {
+        assertThat(map, anEmptyMap()); // Validate method contract
+        map.putAll(Map.of("a", 1, "b", 2, "c", 3, "d", 4));
+
+        Collection<Integer> values = map.values();
+
+        values.remove(3);
+        values.removeIf(k -> k.equals(2));
+        values.retainAll(Set.of(4, 8));
+        assertThat(map, equalTo(Map.of("d", 4)));
+
+        values.clear();
+        assertThat(map, anEmptyMap());
+        assertThrows(UnsupportedOperationException.class, () -> values.add(7));
+    }
+
+    public static void verifyMapEntrySetIsMutable(Map<String, Integer> map) {
+        assertThat(map, anEmptyMap()); // Validate method contract
+        map.putAll(Map.of("a", 1, "b", 2, "c", 3, "d", 4));
+
+        Set<Map.Entry<String, Integer>> entrySet = map.entrySet();
+        entrySet.remove(Map.entry("b", 2));
+        entrySet.remove(Map.entry("d", 2)); // no change
+        entrySet.removeIf(k -> k.equals(Map.entry("b", 2)));
+        entrySet.retainAll(Set.of(Map.entry("d", 4), Map.entry("e", 5)));
+        assertThat(map, equalTo(Map.of("d", 4)));
+
+        Map.Entry<String, Integer> entry = entrySet.iterator().next();
+        entry.setValue(7);
+        assertThat(map, equalTo(Map.of("d", 7)));
+
+        entrySet.clear();
+        assertThat(map, anEmptyMap());
+        assertThrows(UnsupportedOperationException.class, () -> entrySet.add(Map.entry("z", 7)));
+    }
+
+    public static void verifyMapIsMutableBySequencedMapValues(SequencedMap<String, Integer> map) {
```

Comment:
create issue for more map types? NavigableMap extends SequencedMap but will throw on these methods here